src/mpid/ch4/netmod/ofi/ofi_pre.h

```diff
@@ -127,20 +127,18 @@ typedef struct {
 } MPIDI_OFI_am_request_t;
 
 
-typedef struct {
-    void *buf;
-    size_t count;
-    MPI_Datatype datatype;
-    char pack_buffer[];
-} MPIDI_OFI_pack_t;
-
 typedef struct {
     struct fi_context context[MPIDI_OFI_CONTEXT_STRUCTS];       /* fixed field, do not move */
     int event_id;               /* fixed field, do not move */
     int util_id;
     MPI_Datatype datatype;
     union {
-        MPIDI_OFI_pack_t *pack;
+        struct {
+            void *buf;
+            size_t count;
+            MPI_Datatype datatype;
+            char *pack_buffer;
+        } pack;
```

Comment:
This grows the request structure size by 16 bytes on x86, if I understand correctly. In the past, there was concern about going beyond a cacheline boundary. I can do a quick experiment to verify if we are within bounds.

Reply:
Current size of `MPIR_Request` is 324 bytes with `ch4/ofi`, so this increase is no problem.

Reply:
FWIW, the request allocation itself is dynamic (mostly), so the cache size makes little difference here.  In `ch3`, we had a model where the frequently accessed field are packed close together, to get cache benefit, but that made little difference in practice too.  If we'd like to optimize for cache misses, the overall infrastructure needs to be redone to allow for request array allocations first.  Optimizing the request structure content before we do that is unnecessary.